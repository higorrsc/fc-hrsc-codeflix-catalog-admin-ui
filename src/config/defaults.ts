import { ListOptions } from '../types/_shared';

export const initialOptions: ListOptions = {
  page: 1,
  perPage: 10,
  rowsPerPage: [10, 25, 50, 100],
  search: '',
};

export const drawerWidth: number = 240;

export const baseUrl: string = 'http://localhost:8000/api';
