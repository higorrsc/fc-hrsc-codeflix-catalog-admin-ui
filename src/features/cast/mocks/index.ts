export const castMemberResponse = {
  data: [
    {
      id: 'ab7b1108-8346-472c-84cd-605c59760c87',
      name: 'Koch',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'f3dece43-2f3d-4b04-a1cf-183392b058ae',
      name: 'Friesen',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '9720606f-e206-4fdc-96e9-e187774b8511',
      name: 'Harvey',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'c4000ba9-aaca-4789-96da-847cc2dc42ae',
      name: 'Turcotte',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '6cc5abb0-9d6b-4495-83c3-6f16180ccd76',
      name: 'Gottlieb',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '32578f57-fad3-4547-b2e0-a9b5bc8d0285',
      name: 'Pfeffer',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'c0699f71-4037-47cd-9141-d83f57c1fbd9',
      name: 'Dicki',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '3db3fcfc-9a6c-4e8d-954b-6da7238da8b0',
      name: 'Weissnat',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '47f87ea5-6a34-4ace-9f6a-c2ca01d330ff',
      name: "O'Hara",
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'dc8d849a-718f-491c-af0a-0152abed2c57',
      name: 'Rempel',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '1ef538fc-7834-42f2-846a-655e45127f0d',
      name: 'Ondricka',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'ca4f075c-0e29-4919-a14e-42df95da3eb1',
      name: 'Witting',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '59ebc756-25a6-4d9a-8aba-c707d3f0322e',
      name: 'Hegmann',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '94809873-d733-4db1-bc75-35b94ef4fb49',
      name: 'Konopelski',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'f71a1dec-63a8-4e2b-acd7-76fe2baffa85',
      name: 'McKenzie',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
  ],
  links: {
    first: 'http://localhost:8000/api/cast_members?page=1',
    last: 'http://localhost:8000/api/cast_members?page=2',
    prev: null,
    next: 'http://localhost:8000/api/cast_members?page=2',
  },
  meta: {
    current_page: 1,
    from: 1,
    last_page: 2,
    path: 'http://localhost:8000/api/cast_members',
    per_page: 15,
    to: 15,
    total: 30,
  },
};

export const castMemberResponsePage2 = {
  data: [
    {
      id: 'a190dd8b-25b3-40f9-bfe1-eb824c2fb44b',
      name: 'Goyette',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '6f686184-cd76-4e2e-99da-40fc22ef5e23',
      name: 'Stoltenberg',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '1565ec34-ecee-4028-a9fa-feb7ba3b5655',
      name: 'Kuvalis',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '009fea8e-34da-4eab-9a47-5745724eeed0',
      name: 'Jones',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '6af702a0-760f-4661-b513-3083437ae27c',
      name: 'Larson',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'b441feaf-b718-43c0-a7b8-1f9417f60c83',
      name: 'Casper',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '8fec17d6-5e50-4da9-b6ba-25db241af7af',
      name: 'Miller',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '523abf2d-55dd-48f4-b869-93e8b9f5e9b9',
      name: 'Littel',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '4e7a368c-0731-47a8-9c3f-dcc404c15a88',
      name: 'Roberts',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '8de0db5d-6ba2-47d1-92e7-75dc3e83abf1',
      name: 'Ondricka',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'd0c646d7-7cb5-4e60-a2a2-475d0610ea2d',
      name: 'Koss',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '251bbcc6-81e3-4a6a-8415-088f7788b4f8',
      name: 'Feest',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '6c169539-1938-4f46-8bc3-633f184ab8e4',
      name: 'Effertz',
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: 'a3440340-0a91-404b-88e9-684930648242',
      name: 'Crist',
      type: 2,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
    {
      id: '23a77fa7-7352-4000-80aa-de9d723bd854',
      name: "D'Amore",
      type: 1,
      deleted_at: null,
      created_at: '2025-04-22T22:21:23+0000',
      updated_at: '2025-04-22T22:21:23+0000',
    },
  ],
  links: {
    first: 'http://localhost:8000/api/cast_members?page=1',
    last: 'http://localhost:8000/api/cast_members?page=2',
    prev: 'http://localhost:8000/api/cast_members?page=1',
    next: null,
  },
  meta: {
    current_page: 2,
    from: 16,
    last_page: 2,
    path: 'http://localhost:8000/api/cast_members',
    per_page: 15,
    to: 30,
    total: 30,
  },
};
