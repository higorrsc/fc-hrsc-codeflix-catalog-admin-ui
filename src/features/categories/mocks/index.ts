export const categoryResponse = {
  data: [
    {
      id: '9e871d7b-1113-4523-a624-d7f9ad7c2d97',
      name: 'Cornsilk',
      description: 'Dolorem consequatur rem voluptatem facere.',
      is_active: true,
      deleted_at: null,
      created_at: '2025-04-20T16:55:45+0000',
      updated_at: '2025-04-20T16:55:45+0000',
    },
  ],
  links: {
    first: 'http://localhost:8000/api/categories?page=1',
    last: 'http://localhost:8000/api/categories?page=1',
    prev: null,
    next: '',
  },
  meta: {
    current_page: 1,
    from: 1,
    last_page: 1,
    path: 'http://localhost:8000/api/categories',
    per_page: 15,
    to: 1,
    total: 100,
  },
};
