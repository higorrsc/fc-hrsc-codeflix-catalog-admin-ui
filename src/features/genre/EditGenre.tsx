import { useSnackbar } from 'notistack';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { Page } from '../../components/Page';
import { GenreForm } from './components/GenreForm';
import {
  initialState,
  useGetAllCategoriesQuery,
  useGetGenreByIdQuery,
  useUpdateGenreMutation,
} from './genreSlice';
import { Genre } from './types';
import { mapGenreToForm } from './utils';

export const EditGenre = () => {
  const id = useParams<{ id: string }>().id as string;
  const { enqueueSnackbar } = useSnackbar();
  const [isDisabled, setIsDisabled] = useState(false);
  const [genreState, setGenreState] = useState<Genre>(initialState);
  const [updateGenre, status] = useUpdateGenreMutation();
  const { data: categories } = useGetAllCategoriesQuery();
  const { data: genre, isFetching } = useGetGenreByIdQuery({ id });

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setGenreState({ ...genreState, [name]: value });
  };

  // const handleToggle = (e: React.ChangeEvent<HTMLInputElement>) => {
  //   const { name, checked } = e.target;
  //   setGenreState({ ...genreState, [name]: checked });
  // };

  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();

    await updateGenre(mapGenreToForm(genreState));
  }

  useEffect(() => {
    if (genre) {
      setGenreState(genre.data);
    }
  }, [genre]);

  useEffect(() => {
    if (status.isSuccess) {
      enqueueSnackbar('Genre updated successfully!', { variant: 'success' });
      setIsDisabled(true);
    }
    if (status.error) {
      enqueueSnackbar('Error updating genre!', { variant: 'error' });
    }
  }, [status, enqueueSnackbar]);

  return (
    <Page title='Edit Genre'>
      <GenreForm
        genre={genreState}
        categories={categories?.data}
        isLoading={status.isLoading}
        isDisabled={isDisabled || status.isLoading}
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        // handleToggle={handleToggle}
      />
    </Page>
  );
};
