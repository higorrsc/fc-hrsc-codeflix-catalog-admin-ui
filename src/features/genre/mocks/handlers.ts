import { http, HttpResponse } from 'msw';
import { genreResponse, genreResponsePage2 } from '.';
import { baseUrl } from '../../../config/defaults';

export const handlers = [
  http.get(`${baseUrl}/genres`, ({ request, params, cookies }) => {
    const url = new URL(request.url);
    const page = url.searchParams.get('page');
    if (page === '2') {
      return HttpResponse.json(genreResponsePage2);
    }
    return HttpResponse.json(genreResponse);
  }),
  http.get(`${baseUrl}/genres/:id`, ({ request, params, cookies }) => {
    const { id } = params;
    if (id !== '56250310-7ff5-4346-a95d-76e26f7d4509') {
      return new HttpResponse(null, { status: 404 });
    }
    const genre = genreResponse.data.find((cat) => cat.id === id);
    return HttpResponse.json({ data: genre });
  }),
  http.delete(`${baseUrl}/genres/:id`, ({ request, params, cookies }) => {
    const { id } = params;
    if (id === '56250310-7ff5-4346-a95d-76e26f7d4509') {
      return new HttpResponse(null, { status: 204 });
    }
  }),
  http.post(`${baseUrl}/genres`, ({ request, params, cookies }) => {
    return new HttpResponse(null, { status: 201 });
  }),
  http.put(
    `${baseUrl}/genres/56250310-7ff5-4346-a95d-76e26f7d4509`,
    ({ request, params, cookies }) => {
      return new HttpResponse(null, { status: 200 });
    }
  ),
];
