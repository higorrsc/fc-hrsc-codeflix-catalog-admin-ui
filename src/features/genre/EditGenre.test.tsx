import { http, HttpResponse } from 'msw';
import { baseUrl } from '../../config/defaults';
import { server } from '../../mocks/server';
import {
  fireEvent,
  renderWithProviders,
  screen,
  waitFor,
} from '../../utils/test-utils';
import { EditGenre } from './EditGenre';
import { genreResponse } from './mocks';

const genreToEditId = genreResponse.data[0].id;
const genreToEditName = genreResponse.data[0].name;
const routePath = '/genres/edit/:id';
const initialRoute = `/genres/edit/${genreToEditId}`;

describe('Test EditGenre page', () => {
  it('should render correctly', async () => {
    const { asFragment } = renderWithProviders(<EditGenre />, {
      initialEntries: [initialRoute],
      routePath: routePath,
    });
    await waitFor(() => {
      const nameInput = screen.getByTestId('name-input');
      expect(nameInput).toHaveValue(genreToEditName);
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should handle submit', async () => {
    renderWithProviders(<EditGenre />, {
      initialEntries: [initialRoute],
      routePath: routePath,
    });
    await waitFor(() => {
      const nameInput = screen.getByTestId('name-input') as HTMLInputElement;
      expect(nameInput.value).toBe(genreToEditName);
    });

    const name = screen.getByTestId('name-input');
    const submit = screen.getByRole('button', { name: /Save/i });

    fireEvent.change(name, { target: { value: 'Test Name' } });
    fireEvent.click(submit);

    await waitFor(() => {
      const successMessage = screen.getByText('Genre updated successfully!');
      expect(successMessage).toBeInTheDocument();
    });
  });

  it('should handle submit error', async () => {
    server.use(
      http.put(
        `${baseUrl}/genres/56250310-7ff5-4346-a95d-76e26f7d4509`,
        ({ request, params, cookies }) => {
          return new HttpResponse(null, {
            status: 500,
          });
        }
      )
    );
    renderWithProviders(<EditGenre />, {
      initialEntries: [initialRoute],
      routePath: routePath,
    });
    await waitFor(() => {
      const nameInput = screen.getByTestId('name-input') as HTMLInputElement;
      expect(nameInput.value).toBe(genreToEditName);
    });

    const name = screen.getByTestId('name-input');
    const submit = screen.getByRole('button', { name: /Save/i });

    fireEvent.change(name, { target: { value: 'Test Name' } });
    fireEvent.click(submit);

    await waitFor(() => {
      const errorMessage = screen.getByText('Error updating genre!');
      expect(errorMessage).toBeInTheDocument();
    });
  });
});
