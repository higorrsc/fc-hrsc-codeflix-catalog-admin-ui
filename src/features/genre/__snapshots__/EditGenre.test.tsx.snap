// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EditGenre page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Edit Genre
          </h4>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-19midj6"
          >
            <form>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-1u8xl8z-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="«r0»"
                        id="«r0»-label"
                      >
                        Name
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                        >
                           *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                          data-testid="name-input"
                          id="«r0»"
                          name="name"
                          required=""
                          type="text"
                          value="Saint Lucia"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-w1u3ce"
                          >
                            <span>
                              Name *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
                >
                  <div
                    class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-o5oj2-MuiAutocomplete-root"
                    data-testid="categories-search"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
                      data-testid="categories-input"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="«r1»"
                        id="«r1»-label"
                      >
                        Categories
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1e3qgpi-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <div
                          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-16yb58i-MuiButtonBase-root-MuiChip-root"
                          data-tag-index="0"
                          role="button"
                          tabindex="-1"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                          >
                            SeaGreen
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-16yb58i-MuiButtonBase-root-MuiChip-root"
                          data-tag-index="1"
                          role="button"
                          tabindex="-1"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                          >
                            PeachPuff
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-16yb58i-MuiButtonBase-root-MuiChip-root"
                          data-tag-index="2"
                          role="button"
                          tabindex="-1"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                          >
                            Gray
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-16yb58i-MuiButtonBase-root-MuiChip-root"
                          data-tag-index="3"
                          role="button"
                          tabindex="-1"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                          >
                            SkyBlue
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-16yb58i-MuiButtonBase-root-MuiChip-root"
                          data-tag-index="4"
                          role="button"
                          tabindex="-1"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                          >
                            SandyBrown
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                        </div>
                        <input
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-invalid="false"
                          autocapitalize="none"
                          autocomplete="off"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-qve9l1-MuiInputBase-input-MuiOutlinedInput-input"
                          id="«r1»"
                          role="combobox"
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <div
                          class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
                        >
                          <button
                            aria-label="Clear"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1x7n7v0-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                            tabindex="-1"
                            title="Clear"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                              data-testid="CloseIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                              />
                            </svg>
                          </button>
                          <button
                            aria-label="Open"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-2y136s-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                            tabindex="-1"
                            title="Open"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                              data-testid="ArrowDropDownIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M7 10l5 5 5-5z"
                              />
                            </svg>
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-w1u3ce"
                          >
                            <span>
                              Categories
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
                >
                  <div
                    class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
                      >
                        <span
                          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked Mui-checked css-zp7cnf-MuiButtonBase-root-MuiSwitch-switchBase"
                          data-testid="is-active-input"
                        >
                          <input
                            aria-label="controlled"
                            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
                            name="is_active"
                            type="checkbox"
                          />
                          <span
                            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
                          />
                        </span>
                        <span
                          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                      >
                        Active
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-10egq61"
                  >
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
                      href="/genres"
                      tabindex="0"
                    >
                      Back
                    </a>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mns5ui-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="submit"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
